### ISSUE TITLE
auto-indent in editor?

### ISSUE THREAD

**Issue by ericscheid** [naturalcrit/homebrewery#1556]
When entering brew text into the editor, in some circumstances (hard to pin down) the editor will auto-insert a <kbd>TAB</kbd> indent on entering a line break.

### Before hitting <kbd>RETURN</kbd>
![two lines need separating..](https://user-images.githubusercontent.com/2605467/129061613-05c871d8-268b-4863-9e14-15607bbbe6ac.png)

### After hitting <kbd>RETURN</kbd>
![second line indented with a tab](https://user-images.githubusercontent.com/2605467/129061453-a1cda02c-9b73-4387-afcc-d8700fcfdf1b.png)

This behaviour is undesirable.

----

One use case for auto-indent would be when continuing an indenting of text, like adding a new line here could be done by simply pressing <kbd>RETURN</kbd>
![a bunch of lines, all indented, ready for more typing](https://user-images.githubusercontent.com/2605467/129061233-5a09a088-1683-4c43-a4f0-8836a3d3cc9c.png)

And also continuing indented sub-lists.

----

Has an obvious impact with v3 brews because code blocks are now rendered as code blocks.

**Comment by ericscheid** [comment]
Doesn't happen when entering lines of plain markdown text. Does happen inside `<blockquote/>` and other html codes.

Seems to be a CodeMirror thing .. auto-indenting when entering code, specifically html code here.
https://codemirror.net/doc/manual.html#option_smartIndent
Default is true.

We could maybe add a line for `smartIndent: false,` somewhere in here: https://github.com/naturalcrit/homebrewery/blob/a48c74b2e714e769c0fe714c189ea785990458f8/shared/naturalcrit/codeEditor/codeEditor.jsx#L41-L65

There might be other CodeMirror specific bits involved though, so requires experimenting.

**Comment by ericscheid** [comment]
See also #1202

**Comment by Gazook89** [comment]
I tested your suggestion, just added `smartIndent : false,`  and it seems to work well.  It stopped the indenting inside `<div> </div>`.  

I thought it would take away smart indenting for things like lists (and nested lists), but it did not.  And interestingly, it keeps the appropriate indent level if you add the indent yourself (with 2 spaces rather than a tab).  

So in the below case, I entered the second div with two leading spaces, and then when I press enter the next line has those two spaces.  I thought we would lose that for sure.  So I think this is a slam dunk, at least until someone can show me why I am dead wrong.

```
{{div1
  {{div2

  }}

}}
```
